### PR TITLE
Implement new navbar design

### LINKED ISSUE #723: implement new page header / navigation design
screenshot of the mockup:

nav_header

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

this closes #723 

#### What's this PR do?

this reimplements the site navigation path for the dashboard, to implement the new design. 

#### Where should the reviewer start?

Review the changes to `Navbar.js` (was `Header.js`, but the mdl component we're using for putting links and stuff up there is also called `Header`, oops).

#### How should this be manually tested?

Make sure nothing is broken!

And it looks good and stuff.

### CODE REVIEW

File: static/js/components/Navbar.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    { label: 'Dashboard', path: '/dashboard', regex: /dashboard/ },
+    { label: 'Users', path: '/users', regex: /users/ }
+  ];
+
+  activeTab: Function = (path: string): number => (
+    this.tabs.findIndex(tab => tab.regex.test(path))
+  );
+
+  userMenu: Function = (): void|React$Element => {
+    const { empty } = this.props;
```
Comment (amir-qayyum-khan) [inline static/js/components/Navbar.js]: why this ``empty`` variable?

Comment (aliceriot) [inline static/js/components/Navbar.js]: this is basically just ported over from the old `Header` component.

basically, if we set `empty` it means that the navbar shouldn't show the user menu. we use it on the terms of service page.

**Comment by amir-qayyum-khan** [comment]
@aliceriot is their tests for this PR?

**Comment by aliceriot** [comment]
I didn't write any since it's a styling change, I suppose I could write some simple thing that checks that the right links show up but I don't think it's super necessary

what do you think?

**Comment by amir-qayyum-khan** [comment]
is this interface is ok?
Just for curiosity why is this ``Users`` tab on the top.  The same contents are in dropdown/profile as well.

screen shot 2016-07-20 at 4 30 16 pm

**Comment by amir-qayyum-khan** [comment]
@aliceriot yes we can have test to check that dropdown and tabs render in page.
Why we need users tab. It seem it will have more users (list) in future?

**Comment by amir-qayyum-khan** [comment]
their is one pix deference between top margin/padding between below to images. i.e dropdown has one pixel less image on the top.

screen shot 2016-07-20 at 4 40 30 pm
screen shot 2016-07-20 at 4 40 35 pm

**Comment by aliceriot** [comment]
@amir-qayyum-khan we have a new design for the dropdown, I'm working on implementing that now, is it ok with you to leave any fiddling with that off for now? other issue: #724

**Comment by aliceriot** [comment]
and yeah, I had `/users` up there for testing purposes (making sure I could move between the tabs and stuff) and forgot to take it out.

I'll let you know when it's ready for review again.

**Comment by aliceriot** [comment]
@amir-qayyum-khan ok I deleted the users thing.

**Comment by amir-qayyum-khan** [comment]
👍

**Pull request merged**